File: back/hsk/src/main/java/com/myplayground/hsk/hanzi/Hanzi.java
package com.myplayground.hsk.hanzi;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import javax.annotation.Generated;
import javax.persistence.*;

@Getter
@Setter
@NoArgsConstructor
@Accessors(chain = true)
@Entity
@Table(name = "hanzi")
public class Hanzi {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String traditional;

    private String simplified;

    private String pinyin;

    private String definitions;

}

File: front/src/app/shared/models/classes/hanzi.ts
export class Hanzi {

    public traditional = '';
    public simplified = '';
    public pinyin = '';
    public translation: string[] = [];

    public constructor(
        traditional: string,
        simplified: string,
        pinyin: string,
        translation: string[],
    ) {
        this.traditional = traditional;
        this.simplified = simplified;
        this.pinyin = pinyin;
        this.translation = translation;
    }

    public toString(): string {
        return `${this.simplified}\n${this.pinyin}\n ${this.translation.join(' ')}`;
    }

}
